pkg/catalog/catalog.go

```diff
@@ -2679,6 +2679,12 @@ func (c *Catalog) CopyEntry(ctx context.Context, srcRepository, srcRef, srcPath,
 	dstEntry.Path = destPath
 	dstEntry.AddressType = AddressTypeRelative
 	dstEntry.PhysicalAddress = c.PathProvider.NewPath()
+
+	if copyMetadata {
```

Comment:
I probably don't have enough context, but I find this confusing:

What's the difference between the `srcEntry.Metadata` and the `metadata` that is passed as a param?

And the bool name is `copyMetadata`, but even when it's `false`, a `metadata` is copied...

Reply:
Yes, I agree it's confusing. I'm changing the bool name to `copySrcMetadata`, is that better IYO?

Reply:
The reason for this mysterious if/else block is that we need to support 3 behaviors:
1. Copy metadata from the source entry (metadata:`nil` ,  copySrcMetadata:`true`)
2. Use new metadata (metadata:`{...}` ,  copySrcMetadata:`false`)
3. Don't use any metadata at all to keep backward compatibility of our api (metadata:`nil` ,  copySrcMetadata:`false`)

Reply:
Yes, this makes much more sense now. Thanks.